rclpy/test/test_parameter_client.py

```diff
@@ -0,0 +1,139 @@
+        rclpy.init(context=self.context)
+        self.client_node = rclpy.create_node('test_parameter_client',
+                                             namespace='/rclpy',
+                                             context=self.context)
+        self.target_node = rclpy.create_node('test_parameter_client_target',
+                                             namespace='rclpy',
+                                             allow_undeclared_parameters=True,
+                                             context=self.context)
+        self.target_node.declare_parameter('int_arr_param', [1, 2, 3])
+        self.target_node.declare_parameter('float/param//', 3.14)
+
+        self.client = AsyncParameterClient(self.client_node, 'test_parameter_client_target')
+        self.executor = SingleThreadedExecutor(context=self.context)
+        self.executor.add_node(self.client_node)
+        self.executor.add_node(self.target_node)
+
+    def tearDown(self):
+        self.executor.shutdown()
+        self.client_node.destroy_node()
+        self.target_node.destroy_node()
+        rclpy.shutdown(context=self.context)
+
+    def test_set_parameter(self):
+        future = self.client.set_parameters([
+            Parameter('int_param', Parameter.Type.INTEGER, 88).to_parameter_msg(),
+            Parameter('string/param', Parameter.Type.STRING, 'hello world').to_parameter_msg(),
+        ])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert len(results.results) == 2
+        res = [i.successful for i in results.results]
+        assert all(res)
+
+    def test_get_parameter(self):
+        future = self.client.get_parameters(['int_arr_param', 'float/param//'])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert len(results.values) == 2
+        assert list(results.values[0].integer_array_value) == [1, 2, 3]
+        assert results.values[1].double_value == 3.14
+
+    def test_list_parameters(self):
+        future = self.client.list_parameters()
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert 'int_arr_param' in results.result.names
+        assert 'float/param//' in results.result.names
+
+    def test_describe_parameters(self):
+        future = self.client.describe_parameters(['int_arr_param'])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert results.descriptors[0].type == 7
+        assert results.descriptors[0].name == 'int_arr_param'
+
+    def test_get_paramter_types(self):
+        future = self.client.get_parameter_types(['int_arr_param'])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
```

Comment:
To avoid an invalid index error:
```suggestion
        assert results is not None
        assert len(results.types) == 1
```